pkgs/by-name/po/polymc/package.nix

```diff
@@ -0,0 +1,115 @@
+{
+  addDriverRunpath,
+  flite,
+  gamemode,
+  glfw,
+  jdk17,
+  jdk21,
+  jdk8,
+  kdePackages,
+  lib,
+  libGL,
+  libX11,
+  libXcursor,
+  libXext,
+  libXrandr,
+  libXxf86vm,
+  libpulseaudio,
+  libusb1,
+  mangohud,
+  mesa-demos,
+  openal,
+  polymc-unwrapped,
+  stdenv,
+  symlinkJoin,
+  udev,
+  vulkan-loader,
+  wayland,
+  xrandr,
+
+  jdks ? [
+    jdk21
+    jdk17
+    jdk8
+  ],
+  enableTweaks ? stdenv.hostPlatform.isLinux,
+  additionalLibs ? [ ],
+  additionalPrograms ? [ ],
+  msaClientID ? null,
+}:
+
+let
+  polymc' = polymc-unwrapped.override {
+    inherit msaClientID;
+    gamemodeSupport = enableTweaks;
+  };
+  mangohud-lib = symlinkJoin {
+    name = "mangohud-lib";
+    paths = [ mangohud.out ];
+    postBuild = ''
+      mv $out/lib/mangohud/* $out/lib
+      rmdir $out/lib/mangohud
+    '';
+  };
+in
+
+symlinkJoin {
+  name = "polymc-${polymc'.version}";
+
+  paths = [ polymc' ];
+
+  nativeBuildInputs = [ kdePackages.wrapQtAppsHook ];
+
+  buildInputs =
+    [ kdePackages.qtbase ]
+    ++ lib.optional (
+      lib.versionAtLeast kdePackages.qtbase.version "6" && stdenv.hostPlatform.isLinux
+    ) kdePackages.qtwayland;
+
+  postBuild = "wrapQtAppsHook";
+
+  qtWrapperArgs =
+    let
+      runtimeLibs =
+        [
+          stdenv.cc.cc.lib
+          glfw
+          openal
+          libpulseaudio
+          wayland
+
+          # glfw
+          libGL
+          libX11
+          libXcursor
+          libXext
+          libXrandr
+          libXxf86vm
+
+          udev # oshi
+
+          vulkan-loader # VulkanMod's lwjgl
+        ]
+        ++ lib.optionals stdenv.hostPlatform.isLinux [
+          flite
+          libusb1
+        ]
+        ++ lib.optionals enableTweaks [
+          gamemode.lib
+          mangohud-lib
+        ]
+        ++ additionalLibs;
+
+      runtimePrograms = [
+        xrandr
+        mesa-demos
```

Comment:
I missed this on my last review; we should probably also add `glxinfo` here. If I recall correctly from writing the flake in the upstream PolyMC repo, some older LWJGL version will invoke the command and fail to launch if the program isn't present.

Reply:
If I recall correctly, `mesa-demos` provides glxinfo.

Reply:
I see. Any reason we're using mesa-demos rather than just the glxinfo package?

Reply:
I never saw a `glxinfo` package, maybe it has been removed since the last time you worked with it? I'm not sure. Using `nix-locate`, I'm only finding `glxinfo` with `virtualgl.out` and `mesa-demos`.